nix/nixos/cardano-db-sync-service.nix

```diff
@@ -195,6 +200,22 @@ in {
 
         ${cfg.restoreSnapshotScript}
 
+        if [[ -n "''${WAIT_FOR_NODE_SYNC:-}" ]]
+        then
+          until [ -S $CARDANO_NODE_SOCKET_PATH ]; do
+            echo Waiting for $CARDANO_NODE_SOCKET_PATH
+            sleep 2
```

Comment:
Given that a typical cardano restore takes ~30 mins, I this sleep can be a lot less aggressive.

Maybe every 20s? that still makes for 90 log lines.